pkg/archive.py

```diff
@@ -162,11 +146,35 @@ def __init__(self,
       self.default_mtime = int(default_mtime)
 
     self.fileobj = None
-    if self.gz:
-      # The Tarfile class doesn't allow us to specify gzip's mtime attribute.
-      # Instead, we manually re-implement gzopen from tarfile.py and set mtime.
-      self.fileobj = gzip.GzipFile(
-          filename=name, mode='w', compresslevel=9, mtime=self.default_mtime)
+    self.compressor_cmd = compressor.split() if compressor else None
+    if self.compressor_cmd:
+      pass
```

Comment:
This would be easier to follow if you put lines 169-173 here instead of the pass. 
if compressor_cmd
  ... do that
elif xz|lzma:
   ..
elif ...

Reply:
Those lines need to run after the if/else chain because the `xz` branch might set `compressor_cmd`. I’ll add a comment to the `pass` unless we find another refactoring (`if not self.compressor_cmd` is more expressive but also not great since it requires an additional level of indentation…maybe a helper function).